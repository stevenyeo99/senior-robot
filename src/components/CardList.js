import Card from "./Card";

const CardList = (props) => {

    const { items } = props;

    return (
       <div>
        { 
            items.map((item, i) => {
                return <Card key={i} id={item.id} name={item.name} email={item.email} /> 
            })
        }
       </div> 
    );
};

export default CardList;