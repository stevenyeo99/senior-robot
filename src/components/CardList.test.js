import { shallow } from 'enzyme';

import CardList from './CardList';

it('test card list component', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'Steven',
            email: 'yeomandeveloper@gmail.com'
        }
    ]

    expect(shallow(<CardList items={mockRobots} />)).toMatchSnapshot();
});