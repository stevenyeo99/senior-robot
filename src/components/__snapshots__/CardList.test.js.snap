// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test card list component 1`] = `
<div>
  <Card
    email="yeomandeveloper@gmail.com"
    id={1}
    key="0"
    name="Steven"
  />
</div>
`;
